cmd/cloud_sql_proxy/cloud_sql_proxy.go

```diff
@@ -536,6 +554,17 @@ func main() {
 		RefreshCfgThrottle: refreshCfgThrottle,
 	}
 
+	if *performConnectivityTestsOnStartup == true {
+		for _, instanceConfig := range cfgs {
+			err := checkInstanceConnectivity(instanceConfig.Instance, instanceConfig.IsPublic, proxyClient);
+			if err != nil {
+				log.Fatal(err)
+			}
+		}	
+	}
+	
+	logging.Infof("Ready for new connections")
```

Comment:
Maybe accept connections while the connectivity test is happening?

And if there's a reason not to do that, explain that in the flag doc.

Reply:
I think the biggest reason I had was that the messaging was confusing since "Ready for new connections" was throwing a lot of customers off, thinking things were good to go.

Without knowing the results of the test, it could be misleading. We could as a compromise allow the connections but not log anything out until we know for sure?

Reply:
Actually, that's what we're doing here looking back I think. We're accepting connections but we don't tell the user until we know for sure. I could revise the messaging though (or maybe even add a new log before and after to make it more clear.)

Reply:
This seems right to me - it makes sense to wait to say "Ready for new connections" until we've definitely reached a stable state.